src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1520,6 +1522,17 @@ LedgerManagerImpl::applyTransactions(
 
     prefetchTransactionData(txs);
 
+    std::optional<std::vector<TransactionResultPair>::const_iterator>
+        expectedResultsIter = std::nullopt;
+    if (mApp.getConfig().CATCHUP_SKIP_KNOWN_RESULTS && expectedResults)
```

Comment:
Is it a valid scenario when CATCHUP_SKIP_KNOWN_RESULTS=true but expectedResults is not set?

Reply:
If the `CATCHUP_SKIP_KNOWN_RESULTS=true`, but for some reason we do not have expectedResults (e.g. if for some reason they aren't in the archive), catchup will proceed but no transactions will be skipped. I think I should probably add a warning to inform users that this is the case, but I'm hesitant to report an error in that case as its not fatal. What do you think?

Reply:
I see, this makes me realize: with the current implementation of catchup, won't it fail if results couldn't be downloaded? If so, I think we should make it less strict, so it'll try to download results, but if it can't, catchup should still proceed normally. In this case, I agree the logic here makes sense and we can add a warning (probably not at individual ledger level though as it'll get spammy. We can warn per checkpoint).

Reply:
I changed `WorkSequence` to take an vector of "optional" works, which, if they fail do not block the execution of subsequent works. This step now will log a warning every checkpoint if the downloading of results fails, but catchup will proceed.